### ISSUE TITLE
SA1137 is not raised in C# 12 collection expressions

### ISSUE THREAD

**Issue by ar-schraml** [DotNetAnalyzers/StyleCopAnalyzers#3904]
The error code [`SA1137: Elements should have the same indentation`](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1137.md)  is not raised in C# 12 collection expressions.

Example:
![image](https://github.com/user-attachments/assets/7bbd9ca4-7d6a-49c2-acfd-5d17e15f9ffb)
SA1137 is correctly raised inside an object initialization (_example1_), but it is missing in collection expressions (_example2_)

<details><summary>Example code</summary>
<p>

```c#
    private readonly List<string> example1 = new()
    {
        "a",
"b",
                "c",
    };

    private readonly List<string> example2 =
    [
        "a",
"b",
                "c",
    ];
```

</p>
</details>

**Comment by Mivee** [comment]
Heyha, I'd like to give it a shot, but since its also my first open source contribution in general I dont know how long it will take.
